application-favorites-filter-api/src/main/java/org/xwiki/contrib/favorites/filter/EntityFavoriteFilter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.favorites.filter;
+
+import org.xwiki.filter.FilterEventParameters;
+import org.xwiki.filter.FilterException;
+import org.xwiki.filter.annotation.Default;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.user.UserReference;
+
+/**
+ * Favorite related events.
+ * @since 1.4.0
+ * @version $Id$
+ */
+public interface EntityFavoriteFilter
```

Comment:
It's not very clear to me why you have this "Entity" prefix, but it might be more obvious for those who know the favorite extension well.